core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/client/ClientScanner98.java.tmpl

```diff
@@ -0,0 +1,490 @@
+   * @throws IOException
+   */
+  public ClientScanner98(final Configuration conf, final Scan scan, final TableName tableName,
+      HConnection connection, RpcRetryingCallerFactory rpcFactory,
+      RpcControllerFactory controllerFactory) throws IOException {
+      if (LOG.isTraceEnabled()) {
+        LOG.trace("Scan table=" + tableName
+            + ", startRow=" + Bytes.toStringBinary(scan.getStartRow()));
+      } 
+      this.scan = scan;
+      this.tableName = tableName;
+      this.lastNext = System.currentTimeMillis();
+      this.connection = connection;
+      if (scan.getMaxResultSize() > 0) {
+        this.maxScannerResultSize = scan.getMaxResultSize();
+      } else {
+        this.maxScannerResultSize = conf.getLong(
+          HConstants.HBASE_CLIENT_SCANNER_MAX_RESULT_SIZE_KEY,
+          HConstants.DEFAULT_HBASE_CLIENT_SCANNER_MAX_RESULT_SIZE);
+      }
+      this.scannerTimeout = HBaseConfiguration.getInt(conf,
+        HConstants.HBASE_CLIENT_SCANNER_TIMEOUT_PERIOD,
+        HConstants.HBASE_REGIONSERVER_LEASE_PERIOD_KEY,
+        HConstants.DEFAULT_HBASE_CLIENT_SCANNER_TIMEOUT_PERIOD);
+
+      // check if application wants to collect scan metrics
+      initScanMetrics(scan);
+
+      // Use the caching from the Scan.  If not set, use the default cache setting for this table.
+      if (this.scan.getCaching() > 0) {
+        this.caching = this.scan.getCaching();
+      } else {
+        this.caching = conf.getInt(
+            HConstants.HBASE_CLIENT_SCANNER_CACHING,
+            HConstants.DEFAULT_HBASE_CLIENT_SCANNER_CACHING);
+      }
+
+      this.caller = rpcFactory.<Result[]> newCaller();
+      this.rpcControllerFactory = controllerFactory;
+
+      initializeScannerInConstruction();
+    }
+
+    protected void initializeScannerInConstruction() throws IOException{
+      // initialize the scanner
+      nextScanner(this.caching, false);
+    }
+
+    protected HConnection getConnection() {
+      return this.connection;
+    }
+
+    /**
+     * @return Table name
+     * @deprecated Since 0.96.0; use {@link #getTable()}
+     */
+    @Deprecated
+    protected byte [] getTableName() {
+      return this.tableName.getName();
+    }
+
+    protected TableName getTable() {
+      return this.tableName;
+    }
+
+    protected Scan getScan() {
+      return scan;
+    }
+
+    protected long getTimestamp() {
+      return lastNext;
+    }
+
+    // returns true if the passed region endKey
+    protected boolean checkScanStopRow(final byte [] endKey) {
+      if (this.scan.getStopRow().length > 0) {
+        // there is a stop row, check to see if we are past it.
+        byte [] stopRow = scan.getStopRow();
+        int cmp = Bytes.compareTo(stopRow, 0, stopRow.length,
+          endKey, 0, endKey.length);
+        if (cmp <= 0) {
+          // stopRow <= endKey (endKey is equals to or larger than stopRow)
+          // This is a stop.
+          return true;
+        }
+      }
+      return false; //unlikely.
```

Comment:
I did not try to understand or alter the ClientScanner code. So I don't know about this comment. This code comes from earlier version of ClientScanner Hbase, just renamed 98 and adding missing overrides to comply with newer abstract class in hbase 1.0